src/Rendering/prePassRenderer.ts

```diff
@@ -413,90 +476,156 @@ export class PrePassRenderer {
-            for (let i = 0; i < this._scene.activeCamera._postProcesses.length; i++) {
-                if (this._scene.activeCamera._postProcesses[i]?.getClassName() === "ImageProcessingPostProcess") {
-                    isIPPAlreadyPresent = true;
+                    if (this._effectConfigurations[j].postProcess) {
+                        this.renderTargets[i]._beforeCompositionPostProcesses.push(this._effectConfigurations[j].postProcess!);
+                    }
                 }
             }
-
         }
 
-        if (!isIPPAlreadyPresent && !this.disableGammaTransform) {
-            this._postProcesses.push(this.imageProcessingPostProcess);
-        }
-        this._bindPostProcessChain();
+        this._reinitializeAttachments();
         this._setState(true);
+        this._updateGeometryBufferLayout();
     }
 
     private _disable() {
         this._setState(false);
+
+        for (let i = 0; i < this.renderTargets.length; i++) {
+            this._setRenderTargetState(this.renderTargets[i], false);
+        }
+
         this._resetLayout();
 
         for (let i = 0; i < this._effectConfigurations.length; i++) {
             this._effectConfigurations[i].enabled = false;
         }
     }
 
-    private _resetLayout() {
-        for (let i = 0 ; i < this._textureFormats.length; i++) {
-            this._textureIndices[this._textureFormats[i].type] = -1;
+    private _getPostProcessesSource(prePassRenderTarget: PrePassRenderTarget, camera?: Camera) : Nullable<PostProcess>[] {
+        if (camera) {
+            return camera._postProcesses;
+        } else if (prePassRenderTarget.renderTargetTexture) {
+            if (prePassRenderTarget.renderTargetTexture.useCameraPostProcesses) {
+                const camera = prePassRenderTarget.renderTargetTexture.activeCamera ? prePassRenderTarget.renderTargetTexture.activeCamera : this._scene.activeCamera;
+                return camera ? camera._postProcesses : [];
+            } else if (prePassRenderTarget.renderTargetTexture.postProcesses) {
+                return prePassRenderTarget.renderTargetTexture.postProcesses;
+            } else {
+                return [];
+            }
+        } else {
+            return this._scene.activeCamera ? this._scene.activeCamera._postProcesses : [];
+        }
+    }
+
+    private _setupOutputForThisPass(prePassRenderTarget: PrePassRenderTarget, camera?: Camera) {
+        // Order is : draw ===> prePassRenderTarget._postProcesses ==> ipp ==> camera._postProcesses
+        const secondaryCamera = camera && this._scene.activeCameras && !!this._scene.activeCameras.length && this._scene.activeCameras.indexOf(camera) !== 0;
+        this._postProcessesSourceForThisPass = this._getPostProcessesSource(prePassRenderTarget, camera);
+        this._postProcessesSourceForThisPass = (this._postProcessesSourceForThisPass.filter((pp) => { return pp != null; }));
+        this._scene.autoClear = true;

```

Comment:
Shouldn't `scene.autoClear` be saved when enabling the pre pass renderer and restored when disabling it to make sure we don't interfere when the pre pass renderer is not used?

Reply:
That's a fair question. In `DefaultRenderingPipeline.ts`, this flag is also hijacked. Maybe @sebavan has some insight ?